Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -19,120 +19,151 @@ class VirtualMachineSlot: NSObject, ObservableObject {
 
     enum Status: Sendable {
         case empty
-        case starting(LaunchConfiguration)
-        case running(LaunchConfiguration, IPv4Address)
-        case stopping
+        case starting(LaunchConfiguration, Task<ManagedVirtualMachine, Error>)
+        case running(ManagedVirtualMachine, IPv4Address)
+        case stopping(LaunchConfiguration)
         case crashed(Error)
     }
 
+    enum Errors: Error {
+        /// the slot was asked to start a new VM when it wasn't available
+        case invalidStartState
+        /// the VM was stopped while starting
+        case vmStartCancelled
+    }
+
     struct ManagedVirtualMachine {
         let machine: VZVirtualMachine
         let config: LaunchConfiguration
+        let ip: IPv4Address
+
+        var handle: String {
+            config.handle
+        }
     }
 
     private let vmManager = VMManager()
 
     @Published
-    var managedVirtualMachine: ManagedVirtualMachine?
-
-    @Published @MainActor
```

Comment:
The class is `@MainActor` so we should be able to remove it from all the properties and functions.